SIPS/sip-16.md

```diff
@@ -0,0 +1,131 @@
+These transaction insights can also specify a transaction severity level to provide extra friction in the transaction confirmation. 
+To expand on this feature, this SIP allows the community to build Snaps that provide arbitrary "insights" into signatures in a similar manner.
+These insights can then be displayed in the MetaMask UI alongside any information provided by MetaMask itself.
+Signature insight Snaps can also return a `severity` key to indicate the severity level of the content being returned. 
+This key uses the same `SeverityLevel` enum used by transaction insight Snaps. 
+
+## Specification
+
+> Formal specifications are written in Typescript. Usage of `CAIP-N` specifications, where `N` is a number, are references to [Chain Agnostic Improvement Proposals](https://github.com/ChainAgnostic/CAIPs).
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" written in uppercase in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Definitions
+
+> This section is non-normative, and merely recapitulates some definitions from [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md).
+
+- `ChainId` - a [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) string.
+  It identifies a specific chain among all blockchains recognized by the CAIP standards. The ChainId refers to the network that the signing account is connected to. It is not necessarily the network where the signed message will be broadcast. 
+  - `ChainId` consists of a `Namespace` and a `Reference`
+    - `Namespace` - a class of similar blockchains. For example EVM-based blockchains.
+    - `Reference` - a way to identify a concrete chain inside a `Namespace`. For example Ethereum Mainnet or one of its test networks.
+- `Method` - a string which can be any of the following 5 [signing methods](https://docs.metamask.io/wallet/concepts/signing-methods/) supported by the MetaMask wallet API. The set of possible signature types may expand in the future as new signing methods are supported. The signing method is required by MetaMask for the dapp to make a signature request. It is not part of the signature payload.
+  1. `eth_signTypedData_v4`
+  2. `eth_signTypedData_v3`
+  3. `eth_signTypedData_v1`
+  4. `personal_sign`
+  5. `eth_sign`
+
+### Snap Manifest
+
+This SIP specifies a permission named `endowment:signature-insight`.
+The permission grants a Snap read-only access to raw signature payloads, before they are accepted for signing by the user.
+
+The permission is specified as follows in `snap.manifest.json` files:
+
+```json
+{
+  "initialPermissions": {
+    "endowment:signature-insight": {}
+  }
+}
+```
+
+The permission includes an OPTIONAL caveat `allowSignatureOrigin`. 
```

Comment:
```suggestion
The permission includes an OPTIONAL caveat `allowOrigin`. 
```

Reply:
OK but the caveat for transaction insights is `allowTransactionOrigin` so for consistency I figured this one would be `allowSignatureOrigin`

Reply:
Ok cool